dashboard/modules/serve/serve_agent.py

```diff
@@ -74,6 +74,36 @@ async def get_all_deployments(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    @routes.get("/api/serve/applications/")
+    @optional_utils.init_ray_and_catch_exceptions()
+    async def get_serve_instance_details(self, req: Request) -> Response:
+        from ray.serve.schema import ServeInstanceDetails
+
+        controller = await self.get_serve_controller()
+
+        if controller is None:
+            # If no serve instance is running, return a dict that represents that.
+            details = ServeInstanceDetails.get_empty_schema_dict()
+        else:
+            try:
+                details = await controller.get_serve_instance_details.remote()
+            except ray.exceptions.RayTaskError as e:
+                # Task failure sometimes are due to GCS
+                # failure. When GCS failed, we expect a longer time
+                # to recover.
+                return Response(
+                    status=503,
+                    text=(
+                        "Fail to get the response from the controller. "
+                        f"Potentially the GCS is down: {e}"
```

Comment:
```suggestion
                        f"The GCS may be down: {e}"
```

Reply:
Can you add an actionable comment to this error (e.g. something like "please retry later")?